Lib/functools.py

```diff
@@ -187,15 +187,20 @@ def _lt_from_ge(self, other, NotImplemented=NotImplemented):
 
 def total_ordering(cls):
     """Class decorator that fills in missing ordering methods"""
-    # Find user-defined comparisons (not those inherited from object).
-    roots = {op for op in _convert if getattr(cls, op, None) is not getattr(object, op, None)}
+    # Find user-defined comparisons (not those inherited from object or abstract).
+    roots = {op for op in _convert
+             if getattr(cls, op, None) is not getattr(object, op, None)
+             and not getattr(getattr(cls, op, None), '__isabstractmethod__', False)}
```

Comment:
The nested *getattr()* isn't very clear.  

Also, since this is in the inner loop for a dynamic decorator, it will slow the class creation process.  I didn't used to think that was important but experience with named tuples show that people care a great deal about the amount of computation that occurs during an import.

Reply:
would using a walrus operator be okay here?

```python
roots = {...
         if root:=getattr(cls, op, None) is not getattr(object, op, None)
         and not getattr(root, '__isabstractmethod__', False)}
```

Reply:
That seems fine to me.